### DISCUSSION TITLE
How to simply update the specific network(s) in a circuit object?

### DISCUSSION THREAD

**Discussion by Asachoo** [scikit-rf/scikit-rf#1076] [Q&A] [answered]
Hi,

If I create a Circuit object, how should I efficiently update certain specified Networks in the circuit to obtain the modified circuit and its external parameters?

Such requirements are common, such as optimizing some capacitors, inductors and other components in the circuit, to make the circuit's performance better.

Using explicit code in the notebook to repeatedly update the network and build the circuit object is not a general method.

I believe it would be more useful if a method like `circuit.update(networks)` could be supported in the `circuit` instance. The `networks` parameters here could be a sequence of networks, thereby updating the networks with the same name in the `circuit` instance.

Any advices will be greatly appreciated!

**Comment by jhillairet** [comment, marked as answer]
What I do currently is to create a`Circuit` from specific parameters (like capacitances, lengths, etc). Each time one of the parameter is changed, my `Circuit` shall be updated (recreated actually), but this is fast (it's basically a Python list). What can be slower is the (re)evaluation of the `Network` from the updated `Circuit`. 
I'm afraid you can't avoid recalculating the full resulting Network if any of the networks of the Circuit has been changed...

**Reply by Asachoo** [reply]
Thanks for looking into this, that helped.

**Reply by Asachoo** [reply]
For reference https://github.com/scikit-rf/scikit-rf/pull/1160
